### ISSUE TITLE
Flag posts_per_page value of -1

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#364]
## What problem would the enhancement address for VIP?
On VIP, we strongly recommend not having no limit queries, as they are not scalable on larger sites: https://lobby.vip.wordpress.com/wordpress-com-documentation/code-review-what-we-look-for/#no-limit-queries

When we moved over the deprecated WPCS VIP sniffs over to VPCS [here](https://github.com/Automattic/VIP-Coding-Standards/pull/323/files), the [former PostsPerPageSniff  on WPCS VIP](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/702f56fd96bb00a6f8a059f41f8d390fabb766b7/WordPress/Sniffs/VIP/PostsPerPageSniff.php#L96 ) became known as `NoPagingSniff` on VIPCS and the checks for `posts_per_page` were removed, as the `WordPress.WP.PostsPerPage.posts_per_page_posts_per_page` sniff was intended to be used instead.  However, at the time, it wasn't realized that the WPCS PostsPerPage sniff did not check for values of `-1`.

## Describe the solution you'd like

1) It would be nice to add this upstream onto WPCS's side: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1638

2) But if not for 1), we should have a VIPCS sniff that flags this.  

## What code should be reported as a violation?

```php
$args = [ 'posts_per_page' => -1 ];
```

## What could should *not* be reported as a violation?

```php
$args = [ 'posts_per_page' => 1 ];
```

## Additional context
Also possibly related: https://github.com/Automattic/VIP-Coding-Standards/issues/29

**Comment by GaryJones** [comment]
After a brief discussion with @jrfnl, the approach may be to refactor the upstream sniff to allow for a _range of values_ to check for, rather than a boolean property for checking against `-1`.

Regardless, no action is going to happen before 2.2.0 is released, so I'm punting to 2.3.0.

**Comment by rebeccahum** [comment]
After talking to @jrfnl, we'll wait for WPCS 3.0 to be released.

**Comment by rebeccahum** [comment]
Another code snippet I'd like to see flagged:

```
$query = new WP_Query( 
    array( 
      'post_type'   => 'module', 
      'post_status' => 'publish', 
      'post_parent' => 0,
      'numberposts' => -1, // Error
    ) 
  );
```

**Comment by GaryJones** [comment]
`numberposts` isn't used with `WP_Query` - see https://developer.wordpress.org/reference/classes/wp_query/#pagination-parameters which shows `posts_per_page` and `nopaging`.

`numberposts` is used with `get_posts()` - see https://developer.wordpress.org/reference/functions/get_posts/#parameters

`get_comments()` and `get_pages()` have a `number` argument, but the default is for all items to be returned, so folks can just leave the `number` out.

**Comment by rebeccahum** [comment]
Er, whoops, fixed. It should be flagged in a `get_posts()` context regardless.

**Comment by GaryJones** [comment]
Closed in #882
